### PR TITLE
Fixing `LitQAEvaluation` bugs: incorrect reward indices, not using LLM's native knowledge

### PR DESCRIPTION
**Pull request by jamesbraza**
This PR:
- Fixes both issues identified in https://github.com/Future-House/paper-qa/issues/693
    1. Using LLM's innate knowledge, which this PR fixes by updating `EVAL_PROMPT_TEMPLATE`
    2. Incorrectly handling options beyond the specified options, which this PR fixes by declaring those as "incorrect"
- Fixes bad `discounted_returns` logic and incorrect reward indices
    - Previously, incorrect answers were given 0.1 reward, and unsure answers were given -1.0 reward
    - Adds test coverage of this, both upon `discounted_returns` and the `TaskDataset`
- Makes the input `distractors` be a `Sequence` to avoid in-place edits, further robust-ification after https://github.com/Future-House/paper-qa/pull/694

Closes https://github.com/Future-House/paper-qa/issues/693

### CODE REVIEW

File: paperqa/litqa.py [hunk 1]
```diff
@@ -77,8 +98,8 @@ class LitQAEvaluation(IntEnum):
     """Possible evaluation results for a LitQA question."""
 
     CORRECT = 0
-    INCORRECT = 1
-    UNSURE = 2
+    INCORRECT = 2
+    UNSURE = 1
```
Comment (sidnarayanan) [inline paperqa/litqa.py:102]: Why rely on indexing instead of e.g. using a mapping?

File: paperqa/prompts.py [hunk 2]
```diff
@@ -99,8 +99,8 @@
 QA_PROMPT_TEMPLATE = "Q: {question}\n\nOptions:\n{options}"
 EVAL_PROMPT_TEMPLATE = (
     "Extract the single letter answer from the following question and answer"
-    "\n\n{qa_prompt}"
-    "\n\n{qa_answer}"
+    "\n\nQuestion: {qa_prompt}"
+    "\n\nAnswer: {qa_answer}"
     "\n\nSingle Letter Answer:"
```
Comment (sidnarayanan) [inline paperqa/prompts.py:104]: I'd suggest something like:
```
"Given the following question and a proposed answer to the question, return the single-letter choice that matches the proposed answer."
"\n\nQuestion: ..."
"\n\nProposed Answer: ..."
```

If I didn't know the context of this method, as a human, I'd find the original prompt unclear.

File: paperqa/litqa.py (hunk 1 above)
Comment (jamesbraza) [reply paperqa/litqa.py]: Nice suggestion, done 👌

File: paperqa/litqa.py [hunk 3]
```diff
@@ -112,6 +138,14 @@ def extract_answer(answer: str) -> str:
             return answer.split()[0][0].upper()
 
         result = extract_answer(text)
+        if (
+            total_options is not None
+            and ord(result[0]) - _CAPITAL_A_INDEX + 1 > total_options
+        ):
+            # The result extracted was not in the options
```
[the remark this answers is not in the document]
Comment (jamesbraza) [reply paperqa/litqa.py:145]: Let's make a test case for this

[the remark this answers is not in the document]
Comment (jamesbraza) [reply paperqa/litqa.py:145]: I added a test case for this. TL;DR, we correct state that answer as `LitQAEvaluation.INCORRECT`, but for the wrong reasons. This is a separate issue, so I made https://github.com/Future-House/paper-qa/issues/709 for this. It's beyond this PR

**Review by mskarlin — APPROVED** [review]

File: paperqa/prompts.py (hunk 2 above)
Comment (jamesbraza) [reply paperqa/prompts.py:104]: Yeah it's a nice suggestion, but making this change breaks two of our test cases. Let's save this for another PR

**Pull request merged**